File: RotateAndReverse.py
'''Rotate array by one in clockwise direction and then reverse it'''
def RotateAndReverse():
    
    count = 1
    ch = "y"
    while(int(count) <= 200):
     
        
        
        if (ch == "y"):
            arr = [0]
            N = raw_input("Enter length of array:")
            d = N
            f = int(N) - 1
            
            for i in range(int(N)-1):
                arr.append(0)
            print("enter array elements one by one")
            for i in range(int(N)):
                arr[i] = input("enter element" + "" + str(i+1))
            print arr
            print ("lenth of array" + " " +str(len(arr)))
                
            for i in range(1):
                    
                temp = arr[int(d)-1]
                for j in range(int(d)-2,-1,-1):
                    arr[j+1] = arr[j]
                arr[0] = temp
            print arr
            for i in range(int(N)/2):
                arr[f],arr[i] = arr[i],arr[f]
                i = int(i)+ 1
                f = int(f) - 1
            print arr
        else:
            exit(0)
        print ("no. of test cases done:" + str(count))
        if (int(count) == 199):
            print "last test case remaining"
        count = count + 1
        ch = raw_input("Do you want to test more? y/n:")
        
        
main()

File: RotateAndReverse.java
/*Rotate array by one in clockwise direction and then reverse it*/
import java.util.Scanner;
import java.util.Arrays;
public class RotateAndReverse {
    public static void main(String args[]) {
        int[] arr;
        int N, d, temp;
        boolean flag = true;
        int count = 1;
        char ch = 'y';
        Scanner scan = new Scanner(System.in);

        while(count<201){
            if (ch == 'y'){
                System.out.println("Enter length of array:");
                N = scan.nextInt();
                d = (N-1);
                arr = new int[N];
                System.out.println("enter array elements one by one");
                for (int i=0;i<N;i++){
                    arr[i] = scan.nextInt();
                }
                System.out.println("entered array");
                System.out.println(Arrays.toString(arr));
                for (int i=0;i<1;i++){
                    temp = arr[N-1];
                    for(int j=N-2;j>=0;j--){
                        arr[j+1] = arr[j];
                    }
                    arr[0] = temp;
                }
                    
            
        
                    
                System.out.println("Rotated Array");
                System.out.println(Arrays.toString(arr));
                for (int i=0;i<(N/2);i++){
                    temp = arr[i];
                    arr[i] = arr[d];
                    arr[d] = temp;
                    d = d - 1;
                }
                System.out.println("Reversed Array");
                System.out.println(Arrays.toString(arr));
                    
                    
                
            }
            else{
                flag = false;
                break;
            }
            System.out.println("No. of test cases done: ");
            System.out.println(count);
            if (count == 199){
                System.out.println("Last test case remaining");
            }                
                        
            count = count + 1;
            System.out.println("Do you wish to continue?");
            ch = scan.next().charAt(0);
            
            
                
            
            
        }
        
    }
}
